novawallet/Modules/NetworkManagement/NetworkDetails/NetworkDetailsInteractor.swift

```diff
@@ -314,27 +307,65 @@ private extension NetworkDetailsInteractor {
     func connectToNodes(of chain: ChainModel) {
         filteredNodes.forEach { connectTo($0, of: chain) }
     }
-    
-    func delete(_ node: ChainNodeModel) {
-        guard currentSelectedNode != node else {
-            return
-        }
-        
-        let saveOperation = repository.saveOperation(
-            { [weak self] in
-                guard let self else { return [] }
 
-                return [chain.removing(node: node)]
-            },
-            { [] }
-        )
-        
-        executeDataOperationWithErrorHandling(
-            saveOperation,
-            onSuccess: { [weak self] in
+    func delete(
+        _ node: ChainNodeModel,
+        for chain: ChainModel
+    ) {
+        let wrapper = nodeDeletionWrapper(node, chain: chain)
+
+        execute(
+            wrapper: wrapper,
+            inOperationQueue: operationQueue,
+            runningCallbackIn: .main
+        ) { [weak self] result in
+            switch result {
+            case .success:
                 self?.nodesConnections[node.url]?.disconnect(true)
                 self?.nodesConnections[node.url] = nil
+            case .failure:
+                self?.presenter?.didReceive(CommonError.dataCorruption)
             }
+        }
+    }
+
+    func nodeDeletionWrapper(
+        _ node: ChainNodeModel,
+        chain: ChainModel
+    ) -> CompoundOperationWrapper<Void> {
+        let operation: (
+            targetOperation: BaseOperation<Void>,
+            dependencies: [BaseOperation<Void>]
+        )
+
+        if node == currentSelectedNode {
+            let chainWithAutoBalance = chain.updatingConnectionMode(for: .autoBalanced)
+
+            let setAutoBalanceOperation = repository.saveOperation(
+                { [chainWithAutoBalance] },
+                { [] }
+            )
+
+            let deleteNodeOperation = repository.saveOperation(
```

Comment:
Didn't get why do we need two operations, why just not have a single operation that saves the updated chain?

Reply:
There is some flaky behavior with single operation approach. Sometimes we don't receive `.connected` event on `WebSocketEngineDelegate` when updating connectionMode with removing node at the same time. 

Fixed, but further investigation and tests required.